### PR TITLE
ZHA network settings and backup/restore interface

### PR DESCRIPTION
**Pull request by puddly**
## Proposed change

Add a new tab to the ZHA interface to show the current network settings to allow them to be backed up/restored.

Unmerged Core PR: https://github.com/home-assistant/core/pull/75791

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

Testing this requires ZHA to be set up with a Zigbee radio. I've included screenshots below of the various interactions (dialogs moved around with dev tools to show more elements at once):

<details>
<summary><strong>Expand screenshots</strong></summary>
image

image

image

</details>

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by frenck** [comment]
Backend has been merged.

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+  showAlertDialog,
+  showConfirmationDialog,
+} from "../../../../../dialogs/generic/show-dialog-box";
+
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+import { haStyle, haStyleDialog } from "../../../../../resources/styles";
+import { HomeAssistant } from "../../../../../types";
+
+import { HaRadio } from "../../../../../components/ha-radio";
+import { HaCheckbox } from "../../../../../components/ha-checkbox";
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts:34]:
```suggestion
import type { HaRadio } from "../../../../../components/ha-radio";
import type { HaCheckbox } from "../../../../../components/ha-checkbox";
import from "../../../../../components/ha-radio";
import from "../../../../../components/ha-checkbox";
```

This makes sure to also import the element itself

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts [hunk 2]
```diff
@@ -0,0 +1,368 @@
+import {
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts:1]: Can you remove all the empty lines in the import section

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+    this._backupFile = undefined;
+    this._backupType = undefined;
+    this._chosenBackup = undefined;
+    this._currentBackups = undefined;
+    this._overwriteCoordinatorIEEE = false;
+
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  private _formatBackupLabel(backup: ZHANetworkBackup): string {
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts:81]: `private` methods need to be moved below any `public` or `protected`

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+    const backupTime = new Date(Date.parse(backup.backup_time));
+    return `${backupTime.toLocaleString()} (${backup.network_info.pan_id} / ${
+      backup.network_info.extended_pan_id
+    }, ${backup.network_info.source})`;
+  }
+
+  private _shouldOverwriteCoordinatorIEEEAddress(): boolean {
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts:88]: same

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+      this._chosenBackup!.node_info.ieee ===
+      this._currentSettings.settings.node_info.ieee
+    ) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private _canOverwriteCoordinatorIEEEAddress(): boolean {
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts:107]: same

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+                      accept=".json,application/json"
+                      label=${this._backupFile?.name ??
+                      this.hass.localize(
+                        "ui.panel.config.zha.network.upload_backup"
+                      )}
+                      @file-picked=${this._uploadFile}
+                    ></ha-file-upload>
+                  </div>
+                `
+              : nothing
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts:212]:
```suggestion
              : ""
```

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+                              name="chosenAutomaticBackup"
+                              @change=${this._handleAutomaticBackupChanged}
+                              .value=${backup}
+                            ></ha-radio>
+                          </ha-formfield>
+                        `
+                      )}
+                    </div>
+                  `
+              : nothing
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts:192]:
```suggestion
              : ""
```

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+                      )}
+                    >
+                      <ha-checkbox
+                        @change=${this._overwriteCoordinatorIEEEChanged}
+                        .checked=${this._overwriteCoordinatorIEEE}
+                      ></ha-checkbox>
+                    </ha-formfield>
+                  </div>
+                `
+              : nothing
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-restore-backup.ts:231]:
```suggestion
              : ""
```

File: src/panels/config/integrations/integration-panels/zha/zha-network-page.ts [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+import "../../../ha-config-section";
+import "../../../../../components/ha-form/ha-form";
+import "../../../../../components/buttons/ha-progress-button";
+import {
+  fetchZHANetworkSettings,
+  createZHANetworkBackup,
+  ZHANetworkSettings,
+  ZHANetworkBackupAndMetadata,
+} from "../../../../../data/zha";
+
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/zha-network-page.ts:30]:
```suggestion
```

**Comment by puddly** [comment]
I believe I've addressed all of the issues you mentioned.

Is there any `pre-commit`-like hook for the frontend that auto-formats everything properly, like there is for HA Core?

**Comment by zsarnett** [comment]
Ill take a look this week to try and test it. First review is usually just a quick glance.

**Comment by Hedda** [comment]
Nice! Please remember that ZHA docs need update for this

**Pull request closed without merging**